src/vs/platform/terminal/node/terminalEnvironment.ts

```diff
@@ -230,6 +236,39 @@ export function getShellIntegrationInjection(
 	return undefined;
 }
 
+/**
+ * On macOS the profile calls path_helper which adds a bunch of standard bin directories to the
+ * beginning of the PATH. This causes significant problems for the environment variable
+ * collection API as the custom paths added to the end will now be somewhere in the middle of
+ * the PATH. To combat this, VSCODE_PATH_PREFIX is used to re-apply any prefix after the profile
+ * has run. This will cause duplication in the PATH but should fix the issue.
```

Comment:
can duplication in the PATH cause issues?

Reply:
It shouldn't cause problems, just a bit ugly if you check the variable.